packages/core/src/pubsub/pubsub.ts

```diff
@@ -48,11 +48,14 @@ function ipfsToPubsub(
  */
 export class Pubsub extends Observable<PubsubMessage> {
   private readonly peerId$: Observable<string>
+  private readonly pubsubKeepaliveInterval
+  private lastPublishedMessageDate: number = Date.now() - this.maxPubsubPublishInterval
```

Comment:
Add another thing on top, please. Too much responsibility in here already. Let's not make it a stuffed turkey.

Reply:
Sorry, I don't actually understand what you're asking for here

Reply:
Proposed changes to `Pubsub` add here one more responsibility - posting a keep-alive message when idle. Current code is complex enough, better handle that responsibility somewhere else. It is proper time now to recall a single-responsibility principle.

Upstream implementation uses `new MessageBus(new Pubsub())` construction. We can wrap a Pubsub instance with `KeepAlive` class that provides the an interface similar to Pubsub (i.e. Observable + next) but also sends a keep-alive message (at most) every N seconds. Like this: https://github.com/ceramicnetwork/js-ceramic/blob/su/keep-alive-wrap/packages/core/src/pubsub/keep-alive.ts; usage: https://github.com/ceramicnetwork/js-ceramic/blob/su/keep-alive-wrap/packages/core/src/dispatcher.ts#L57

Note: few console-logs here and there for visibility purposes.